core/src/Instance/Authority.cs

```diff
@@ -118,27 +118,41 @@ public static void ValidateAsUri(string authority)
             }
         }
 
-        private static Authority CreateInstance(string authority, bool validateAuthority)
+        internal static AuthorityType GetAuthorityType(string authority)
         {
-            authority = CanonicalizeUri(authority);
-            ValidateAsUri(authority);
             string[] pathSegments = new Uri(authority).AbsolutePath.Substring(1).Split('/');
```

Comment:
>new Uri(authority).AbsolutePath.Substring(1).Split('/'); [](start = 36, length = 56)

can you use this: [https://msdn.microsoft.com/en-us/library/system.uri.segments(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/system.uri.segments(v=vs.110).aspx)

Reply:
[https://docs.microsoft.com/en-us/dotnet/api/system.uri.segments?view=netframework-4.7.1#System_Uri_Segments](https://docs.microsoft.com/en-us/dotnet/api/system.uri.segments?view=netframework-4.7.1#System_Uri_Segments)

---
In reply to: [205900836](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1151#discussion_r205900836) [](ancestors = 205900836)

Reply:
switched to Segments api